app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java

```diff
@@ -0,0 +1,81 @@
+package ingsw.group1.findmyphone.activity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.eis.smslibrary.SMSPeer;
+
+import ingsw.group1.findmyphone.R;
+import ingsw.group1.findmyphone.contacts.SMSContactSwipeCallback;
+import ingsw.group1.findmyphone.contacts.SMSContact;
+import ingsw.group1.findmyphone.contacts.SMSContactManager;
+
+/**
+ * Activity for adding a new contact
+ *
+ * @author Giorgia Bortoletti
+ */
+public class ModifyContactActivity extends AppCompatActivity {
+
+    private static String contactAddress;
+
+    private EditText contactNameField;
+
+    private SMSContactManager contactManager;
+    private String contactName;
+
+    /**
+     * Method invoked when activity is created
+     *
+     * @param savedInstanceStatus
+     */
+    @Override
+    public void onCreate(Bundle savedInstanceStatus) {
+        super.onCreate(savedInstanceStatus);
+        setContentView(R.layout.activity_modify_contact);
+
+        contactNameField = findViewById(R.id.modify_contact_name);
+        TextView contactPhoneField = findViewById(R.id.modify_contact_address);
+
+        contactManager = SMSContactManager.getInstance(getApplicationContext());
+
+        SMSContact contact = contactManager.getContactForPeer(new SMSPeer(contactAddress));
+        contactName = contact.getName();
+
+        contactNameField.setText(contactName);
+        contactPhoneField.setText(contactAddress);
+
+    }
+
+    /**
+     * Called when the user touches the button for modifying an existing contact
+     *
+     * @param view {@link View} where this method is invoked
+     */
+    public void modifyContact(View view) {
+        String newContactName = contactNameField.getText().toString();
+
+        if (newContactName.equals(contactName)) {
+            Toast.makeText(getApplicationContext(), ActivityConstantsUtils.EQUALS_CONTACT_NAME, Toast.LENGTH_LONG).show();
+        } else {
+            contactManager.modifyContactName(new SMSPeer(contactAddress), newContactName);
+            Toast.makeText(getApplicationContext(), ActivityConstantsUtils.CONTACT_UPDATED, Toast.LENGTH_LONG).show();
+        }
+        startActivity(new Intent(ModifyContactActivity.this, ContactListActivity.class));
```

Comment:
You don't need to start a new activity, it's already open behind this, you should be able to call `onBackPressed` and obtain the same result

Reply:
It's a very good advice. I changed that lines but the problem remains: when `ModifyContactActivity` is closed, I need to update the list of contacts in the `ContactListActivity`. So I have implemented the `onResume` method and notified changes at the `Adapter`. But it doesn't refresh data. Do you have some advice?

Reply:
Ok I solved it. Now I do a _commit_ with changes.